test/e2e/agent/agent_updates_test.go

```diff
@@ -75,6 +75,80 @@ var _ = Describe("VM Agent behavior during updates", func() {
 			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
 			logrus.Info("Device updated to new image 🎉")
 		})
+
+		It("Should update to v4 with embedded application", Label("updates", "77667"), func() {
+			// Check the device status right after bootstrap
+			response := harness.GetDeviceWithStatusSystem(deviceId)
+			device := response.JSON200
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+			Expect(*device.Status.Summary.Info).To(Equal("Bootstrap complete"))
+			Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+			var newImageReference string
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":v4"
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Update")
+				}, "1m")
+
+			harness.WaitForDeviceContents(deviceId, "the device is rebooting",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "True", "Rebooting")
+				}, "2m")
+
+			harness.WaitForDeviceContents(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference &&
+						conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+
+			// TODO(hexfusion): we were expecting this update status not to be unknown at this point
+			// related to: https://issues.redhat.com/browse/EDM-679
+			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+			logrus.Infof("Device updated to new image %s 🎉", "flightctl-device:v4")
+			logrus.Info("Pods with sleep infinity process are present but not running, as expected")
```

Comment:
```suggestion
			logrus.Info("We expect containers with sleep infinity process to be present but not running")
```

Feel free to fix later. This is just logging, but it's better to be clear.

Reply:
will change

Reply:
This came back maybe from rebase @Samudelacruz77

Reply:
Yes, will change again